### PR TITLE
OIDC verifier package

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Changes proposed in this pull request:

- oidc package for verification oidc tokens in oci-image-builder ado pipeline.
- Package abstracts external dependencies.
- Package provides a provider created using OIDC discovery mechanism.
- Package provides a verifier for oidc tokens.
- Package provides a token processor as a type to work with tokens.
- Package extends standard token claims with Github specific claims.
- Package is tested using mocks generated by mockery and ginkgo, gomega framework.
- The package is intended to use in short living ADO pipelines and validate only one token at the time. This is not a server implementation.
- Package contains only debug severity messages for troubleshooting purposes.
- The logging uses interfaces from github.com/kyma-project/test-infra.

**Related issue(s)**
See: https://github.com/orgs/kyma-project/projects/22/views/70?pane=issue&itemId=56193547

### CODE REVIEW

File: pkg/oidc/oidc.go [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+}
+
+// Issuer returns the issuer of the token.
+func (tokenProcessor *TokenProcessor) Issuer() string {
+	return tokenProcessor.issuer
+}
+
+// Claims verify and parse the token to get the token claims.
+// It unmarshal the claims into the provided claims struct.
+func (tokenProcessor *TokenProcessor) Claims(ctx context.Context, verifier TokenVerifierInterface, claims ClaimsInterface) error {
```
Comment (akiioto) [inline pkg/oidc/oidc.go:338]: Change the name of this function to sth like, "VerifyAndExtractClaims".

Comment (dekiel) [reply pkg/oidc/oidc.go]: renamed

**Review by akiioto — APPROVED** [review]

**Pull request merged**